source/reflectionzeug/source/JsonFromVariantSerializer.cpp

```diff
@@ -0,0 +1,172 @@
+#include <reflectionzeug/JsonFromVariantSerializer.h>
+
+#include <memory>
+#include <ostream>
+#include <fstream>
+#include <sstream>
+
+#include <loggingzeug/logging.h>
+
+
+using namespace loggingzeug;
+
+namespace reflectionzeug
+{
+    
+JsonFromVariantSerializer::JsonFromVariantSerializer()
+    : m_nestingLevel(0)
+{
+}
+
+JsonFromVariantSerializer::~JsonFromVariantSerializer()
+{
+}
+
+void JsonFromVariantSerializer::writeToStream(Variant & variant, std::ostream & outStream)
+{
+    serialize(variant, outStream);
+}
+
+void JsonFromVariantSerializer::writeToFile(Variant & variant, const std::string & filePath)
+{
+    auto fileStream = std::unique_ptr<std::ofstream>(new std::ofstream(filePath));
+    if (fileStream->is_open())
+    {
+        serialize(variant, *fileStream);
+    }
+    else
+    {
+        critical() << "Could not open stream. Aborted writing to file: \"" << filePath << "\".";
+    }
+}
+
+std::string JsonFromVariantSerializer::writeToString(Variant & variant)
+{
+    auto stringStream = std::unique_ptr<std::ostringstream>(new std::ostringstream());
+    serialize(variant, *stringStream);
+    return stringStream->str();
+}
+
+void JsonFromVariantSerializer::serialize(Variant & variant, std::ostream & stream)
+{
+    m_nestingLevel = 0;

```

Comment:
This is why you don't need the constructor, you have to set `m_nestingLevel` here anyway.